app/questionnaire/questionnaire_schema.py

```diff
@@ -1248,6 +1254,22 @@ def get_item_anchor(self, section_id: str, list_name: str) -> str | None:
                 if item["for_list"] == list_name and item.get("item_anchor_answer_id"):
                     return f"#{str(item['item_anchor_answer_id'])}"
 
+    def update_dependencies_for_dynamic_answers(
+        self, *, question: Mapping, block_id: str
+    ) -> None:
+        if dynamic_answers := question.get("dynamic_answers"):
```

Comment:
in this if statement, if you did `self.dynamic_answers_parent_block_ids.add(block_id)`
I think you would then not need the `_populate_dynamic_answers_parent_block_ids` method

Reply:
Added now, thanks!